port/win/env_win.cc

```diff
@@ -1245,7 +1244,7 @@ class WinEnv : public Env {
     UniqueCloseHandlePtr fileGuard(hFile, CloseHandleFunc);
 
     // CAUTION! This will map the entire file into the process address space
-    if (options.use_mmap_reads && sizeof(void*) >= 8) {
```

Comment:
The reason this check is here is because there is, in  fact, a potential problem. The configuration options are not explicit enough to omit this check. If you choose to go this route, here is what I suggest:

- Provide a replacement for this run-time check so the size of the file mapped to memory does not exceed what is available. It is very easy to exceed 4Gb addressable limit with an SST size. In practice, though, that limit is less 3.25 Gb on a 32-bit Windows system and that does not count the executable code and other data. Runtime checks will provide an error message instead of just dying.

- Add an explicit comment to the options.h for the memory map access stating this limitation for 32-bit builds.

- Whatever you choose to do your change better be consistent across the platforms.  Here is the relevant code for POSIX and it is the reason the code was written this way on Windows. https://github.com/facebook/rocksdb/blob/master/util/env_posix.cc#L176